registry.go

```diff
@@ -0,0 +1,112 @@
+package main
+
+import (
+	"fmt"
+	"log"
+	"net/http"
+	"os/exec"
+
+	"github.com/giantswarm/microerror"
+)
+
+type RegistryConfig struct {
+	Client *http.Client
+
+	Host         string
+	Organisation string
+	Password     string
+	Token        string
+	Username     string
+}
+
+type Registry struct {
+	client *http.Client
+
+	host         string
+	organisation string
+	password     string
+	token        string
+	username     string
+}
+
+func NewRegistry(cfg *RegistryConfig) (*Registry, error) {
+	if cfg.Client == nil {
+		return nil, microerror.Maskf(invalidConfigError, "%T.Client must not be empty", cfg)
```

Comment:
I like that we're aligning retagger with microerror and the rest of our tooling.